analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpNavigationHelper.cs

```diff
@@ -77,17 +77,7 @@ public static IEnumerable<SwitchSectionSyntax> GetPrecedingSections(this SwitchS
 
         #endregion Switch
 
-        #region Statement
-
-        public static StatementSyntax GetPrecedingStatement(this StatementSyntax currentStatement)
-        {
-            var statements = currentStatement.Parent.ChildNodes().OfType<StatementSyntax>().ToList();
-
-            var index = statements.IndexOf(currentStatement);
-
-            return index == 0 ? null : statements[index - 1];
-        }
-
-        #endregion Statement
+        public static StatementSyntax GetPrecedingStatement(this StatementSyntax currentStatement) =>
+            currentStatement.Parent.ChildNodes().OfType<StatementSyntax>().TakeWhile(x => x != currentStatement).LastOrDefault();
```

Comment:
I don't have data at hand for this change, but I know that link can sometimes be more expensive than non-LINQ queries

you could probably do some simple benchmark in LinqPad of how LINQ vs non-LINQ versions behave

Reply:
Linq has 50% improvement over the expensive `IndexOf` call. 
I just measured it with 10000 repetitions for 10 and 100 statements, starting from middle.